### PR TITLE
String extension to convert strings to integers

### LINKED ISSUE #80: String extension to convert strings to integers
**Requirement**

Add a new string extension to easily convert a string to an integer.
- Add another extension to check if a string is numeric (return a bool)
- Throw an exception when trying to convert a non-numeric string to int

**Value Proposition**

Adding this extension would make it simpler to check if a string is numeric and convert it to a integer. The standard approach is to use the static methods on `int`. Having methods on `string` might be more intuitive.

---

For example, this:

```csharp
bool isInt = int.TryParse(myString, out int _);
```

Becomes:

```csharp
bool isInt = myString.IsInt();
```

And this:

```csharp
int myInt = int.Parse(myString);
```

Becomes:

```csharp
int myInt = myString.ToInt();
```

**Design Ideas**

Add the following extensions on `string`:
- `IsInt()`, returns true if the string can be converted to an int
- `ToInt()`, attempts to convert the string to an int - throws an exception if it fails

Note that it is possible to add more methods like this for other types (e.g. `long` and `double`) but this issue is only for `int`.

### PR DESCRIPTION
**Pull request by ilusi0n**
## Description

This closes https://github.com/ColmBhandal/CsharpExtras/issues/80

## TODO

 - [x] Closes https://github.com/ColmBhandal/CsharpExtras/issues/80
 - [x] Tests added
 - [x] Update CHANGELOG

### CODE REVIEW

**Comment by ilusi0n** [comment]
Hello!
It's been a long time :D
Can you help review it? Also any tips about what I can add to changelog? Thanks!

File: CsharpExtrasTest/Extensions/StringExtensionTest.cs [hunk 1]
```diff
@@ -245,6 +246,53 @@ public void GivenStringWhenStringValueConvertedToIntGreaterThanOrEqualToZeroCall
...
+            //Arrange
+            string value = "0";
+            //Act
+            bool result = value.IsInt();
+            //Assert
+            Assert.IsTrue(result);
+        }
+        
+        [Test]
+        [Category("Unit")]
```
Comment (ColmBhandal) [inline CsharpExtrasTest/Extensions/StringExtensionTest.cs:266]: Parametrise this test & include strings that are "almost" integers, to confirm that the logic still returns false e.g.
"1212s123"
"24124a"
" 34324" (note space)
"1234 12"
"_1"

Not sure what happens for values with decimals ending in zero(s) but should be tested:

"1.0"
"2.00"

Again not sure about decimals ending in non zero (are they truncated or does it return false?)

"1.00002"

Test the max digits too e.g. a really long string - does it break the code? E.g. something like:

"1235987345983402759234085702349857342098573245343243298754302958734208954750347850234895750288888888888888888888888888888888888888888888888888888888888888888888888888888888888888888887532954720938570239485732094857023498572304985732094857"

File: CsharpExtrasTest/Extensions/StringExtensionTest.cs [hunk 2]
```diff
@@ -245,6 +246,53 @@ public void GivenStringWhenStringValueConvertedToIntGreaterThanOrEqualToZeroCall
             //Assert
             Assert.IsTrue(result);
         }
+        
+        [Test]
+        [Category("Unit")]
```
Comment (ColmBhandal) [inline CsharpExtrasTest/Extensions/StringExtensionTest.cs:251]: Parametrise - try different lengths & maybe decimals - see false case

File: CsharpExtrasTest/Extensions/StringExtensionTest.cs [hunk 3]
```diff
@@ -245,6 +246,53 @@ public void GivenStringWhenStringValueConvertedToIntGreaterThanOrEqualToZeroCall
...
+            string value = "teste";
+            //Act
+            bool result = value.IsInt();
+            //Assert
+            Assert.IsFalse(result);
+        }
+        
+        [Test]
+        [Category("Unit")]
+        public void GivenStringWhenIsAIntegerValueThenReturnStringAsAInteger()
```
Comment (ColmBhandal) [inline CsharpExtrasTest/Extensions/StringExtensionTest.cs]: Parametrise - analogous to the `IsInt` tests

File: CsharpExtrasTest/Extensions/StringExtensionTest.cs [hunk 4]
```diff
@@ -245,6 +246,53 @@ public void GivenStringWhenStringValueConvertedToIntGreaterThanOrEqualToZeroCall
...
+            int expected = 0;
+            //Act
+            int result = value.ToInt();
+            //Assert
+            Assert.AreEqual(expected,result);
+        }
+        
+        [Test]
+        [Category("Unit")]
+        public void GivenStringWhenIsNotAIntegerValueThenThrowsInvalidOperationException()
```
Comment (ColmBhandal) [inline CsharpExtrasTest/Extensions/StringExtensionTest.cs]: Parametrise - analogous to the `IsInt` tests

**Comment by ColmBhandal** [comment]
Thanks @ilusi0n. Code looks good. I would add a lot more test cases - easy enough to do if you parametrise the tests with [`TestCase`](https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcase.html).

Also please update the chanelog, thanks!

**Review by ColmBhandal — CHANGES_REQUESTED** [review]
See inline comments

File: CsharpExtrasTest/Extensions/StringExtensionTest.cs (hunk 1 above)
Comment (ilusi0n) [reply CsharpExtrasTest/Extensions/StringExtensionTest.cs:266]: Thanks for the feedback! 

I added the test cases but in the case of " 34324" it seems that the C# ignores that space and converts to int (see here https://docs.microsoft.com/en-us/dotnet/api/system.int32.tryparse?view=net-6.0).

Comment (ColmBhandal) [reply CsharpExtrasTest/Extensions/StringExtensionTest.cs:266]: Interesting!

**Review by ColmBhandal — CHANGES_REQUESTED** [review]
Please add some test cases for negative numbers. I don't think I saw any. If I'm wrong, please correct me.

**Comment by ilusi0n** [comment]
> Please add some test cases for negative numbers. I don't think I saw any. If I'm wrong, please correct me.

No, you are right. I forgot :(

**Review by ColmBhandal — APPROVED** [review]
Looks good, thanks!
